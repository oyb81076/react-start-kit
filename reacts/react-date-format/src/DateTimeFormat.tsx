import moment, { isMoment, Moment } from "moment";
import React from "react";
export interface IDateFormatProps {
  toNow?: boolean;
  date?: string | number | Moment | Date;
  format?: string;
}
export default class DateTimeFormat extends React.Component<IDateFormatProps> {
  public static defaultProps = {
    toNow: false,
    format: "YYYY-MM-DD HH:mm",
  };
  public render() {
    const { format, date, toNow } = this.props;
    if (!date) { return null; }
    const value: Moment = isMoment(date) ? date : moment(date);
    if (toNow) {
      return value.toNow();
    } else {
      return value.format(format);
    }
  }
}
