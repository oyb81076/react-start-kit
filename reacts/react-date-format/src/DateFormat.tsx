import DateTimeFormat from "./DateTimeFormat";
export default class DateFormat extends DateTimeFormat {
  public static defaultProps = {
    ...DateTimeFormat.defaultProps,
    format: "YYYY-MM-DD",
  };
}
