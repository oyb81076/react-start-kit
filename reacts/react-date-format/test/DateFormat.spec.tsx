import moment from "moment";
import React from "react";
import { create } from "react-test-renderer";
import DateFormat from "../src/DateFormat";
import DateTimeFormat from "../src/DateTimeFormat";
const date = new Date("2018-10-07T11:39:10.674Z");
test("null", () => {
  const component = create(<DateFormat date={date} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("default", () => {
  const component = create(<DateFormat date={date} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("toNow", () => {
  const m = moment().add(-1, "h");
  const component = create(<DateFormat toNow date={m} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("datetime", () => {
  const component = create(<DateTimeFormat date={date} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
