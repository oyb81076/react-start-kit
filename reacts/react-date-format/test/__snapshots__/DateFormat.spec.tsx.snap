// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datetime 1`] = `"2018-10-07 19:39"`;

exports[`default 1`] = `"2018-10-07"`;

exports[`null 1`] = `"2018-10-07"`;

exports[`toNow 1`] = `"in an hour"`;
